### ISSUE TITLE
Swagger is not generated with loopback host 0.0.0.0

### ISSUE THREAD

**Issue by DallasO** [completecoding/serverless-auto-swagger#97]
Adding `--host 0.0.0.0` when running serverless (framework v3) prevents auto-swagger from generating files locally.

```
# Will generate successfully
sls offline start

# Will not generate
sls offline --host 0.0.0.0 start
```

Packages:
- node: 16.16.0
- serverless: 3.22.0
- serverless-auto-swagger: 2.8.2
- serverless-offline: 10.3.1

Still working on a minimal reproduction, so I can share that once its available

**Comment by bfaulk96** [comment]
As serverless-auto-swagger doesn't deal with any configuration related to `host`, I would think this would be a bug with `serverless-offline` and perhaps how it handles the various lifecycle hooks.

**Comment by bfaulk96** [comment]
To further help you dig into this one the `serverless-offline` side, `serverless-auto-swagger` makes use of the following two hooks:

```
      'before:offline:start:init': this.preDeploy,
      'before:package:cleanup': this.preDeploy,
```

And `preDeploy` will generate your swagger files if you have `generateSwaggerOnDeploy` set to `true` (or `true` by default), on top of creating the endpoint.

For a temporary workaround, you should be able to run `sls generate-swagger` to manually create the files, assuming `sls offline --host 0.0.0.0 start` at least creates the swagger endpoints

**Comment by DallasO** [comment]
Thank you for your prompt response. I posted this out of frustration and as I though about it more, I came to a similar conclusion that it would make more sense that `serverless-offline` is the culprit.

Thank you for the additional details as well, I'm still learning some of the serverless ecosystem, so this is helpful. `sls offline --host 0.0.0.0 start` does not even create the swagger endpoints for me, but I will dig into this more.
